src/register/EmbedableRegistrationForm.jsx

```diff
@@ -0,0 +1,591 @@
+            fieldError = formatMessage(messages['email.invalid.format.error']);
+          } else {
+            const [username, domainName] = value.split('@');
+            // Check if email address is invalid. If we have a suggestion for invalid email
+            // provide that along with the error message.
+            if (!emailRegex.test(value)) {
+              fieldError = formatMessage(messages['email.invalid.format.error']);
+              setEmailSuggestion({
+                suggestion: getSuggestionForInvalidEmail(domainName, username),
+                type: 'error',
+              });
+            } else {
+              const response = validateEmailAddress(value, username, domainName);
+              if (response.hasError) {
+                fieldError = formatMessage(messages['email.invalid.format.error']);
+                delete response.hasError;
+              }
+              setEmailSuggestion({ ...response });
+
+              if (configurableFormFields.confirm_email && value !== configurableFormFields.confirm_email) {
+                confirmEmailError = formatMessage(messages['email.do.not.match']);
+              }
+            }
+          }
+          break;
+        case 'username':
+          if (!value || value.length <= 1 || value.length > 30) {
+            fieldError = formatMessage(messages['username.validation.message']);
+          } else if (!value.match(/^[a-zA-Z0-9_-]*$/i)) {
+            fieldError = formatMessage(messages['username.format.validation.message']);
+          }
+          break;
+        case 'password':
+          if (!value || !LETTER_REGEX.test(value) || !NUMBER_REGEX.test(value) || value.length < 8) {
+            fieldError = formatMessage(messages['password.validation.message']);
+          }
+          break;
+        case 'country':
+          if (flags.showConfigurableEdxFields || flags.showConfigurableRegistrationFields) {
+            const {
+              countryCode, displayValue, error,
+            } = validateCountryField(value.displayValue.trim(), countryList, formatMessage(messages['empty.country.field.error']));
+            fieldError = error;
+            countryFieldCode = countryCode;
+            setConfigurableFormFields(prevState => ({ ...prevState, country: { countryCode, displayValue } }));
+          }
+          break;
+        default:
+          if (flags.showConfigurableRegistrationFields) {
+            if (!value && fieldDescriptions[fieldName]?.error_message) {
+              fieldError = fieldDescriptions[fieldName].error_message;
+            } else if (fieldName === 'confirm_email' && formFields.email && value !== formFields.email) {
+              fieldError = formatMessage(messages['email.do.not.match']);
+            }
+          }
+          break;
+    }
+    if (setError) {
+      setErrors(prevErrors => ({
+        ...prevErrors,
+        confirm_email: flags.showConfigurableRegistrationFields ? confirmEmailError : '',
```

Comment:
This code is extra and is not needed for embedded registration form.

Reply:
addressed